codec/encoder/core/inc/param_svc.h

```diff
@@ -426,6 +432,11 @@ typedef struct TagWelsSvcCodingParam: SEncParamExt {
       pSpatialLayer->uiTransferCharacteristics = pCodingParam.sSpatialLayers[iIdxSpatial].uiTransferCharacteristics;
       pSpatialLayer->uiColorMatrix =             pCodingParam.sSpatialLayers[iIdxSpatial].uiColorMatrix;
 
+      pSpatialLayer->bAspectRatioPresent = pCodingParam.sSpatialLayers[iIdxSpatial].bAspectRatioPresent;
+      pSpatialLayer->eAspectRatio = pCodingParam.sSpatialLayers[iIdxSpatial].eAspectRatio;
+      pSpatialLayer->sAspectRatioExtWidth = BIG_ENDIAN_16(pCodingParam.sSpatialLayers[iIdxSpatial].sAspectRatioExtWidth);
```

Comment:
1. On little endian machine the bytes must be swapped, or the value we write will be decoded incorrectly (on any system).For example: this code was checked with ffmpeg decoder, and it shows the same width/height that were set in openh264 encoder. This means ffmpeg's decoder swaps bytes after reading width/height (on x86).

The problem of BIG_ENDIAN_16 macro is that it works like x86 system is the only little endian system. It should be something like this

```
#ifdef x86 // or some other little endian arch
#define LITTLE_ENDIAN 1
#define BIG_ENDIAN
#else
#define LITTLE_ENDIAN
#define BIG_ENDIAN 1
#endif

#ifdef LITTLE_ENDIAN
#define BIG_ENDIAN_CAST_16(x) ((x>>8) & (x<<8))
#else
#define BIG_ENDIAN_CAST_16(x) x
#endif
```

2. look at the approximation:
```
set value
while (...) { write value }
```
you can see that if we'll swap bytes on writing, the machine will do the unnecessary work.